src/formatter/step_definition_snippet_builder/snippet_syntax.ts

```diff
@@ -11,7 +11,7 @@ export enum SnippetInterface {
 export interface ISnippetSyntaxBuildOptions {
   comment: string
   functionName: string
-  generatedExpressions: GeneratedExpression[]
+  generatedExpressions: readonly GeneratedExpression[]
```

Comment:
Why was readonly added here?

Reply:
I assume to prevent accidental modification of the array and to signal that the array is not meant to be modified. I tend to use it in most signatures and only use modifiable arrays in implementations.

Reply:
Yep - the `cucumber-expressions` library declares it as a `ReadonlyArray` as of https://github.com/cucumber/cucumber/commit/1b1cde5b59d79e155b8324701c46685a45b0dd2f, so we have to match that here to not get a compiler error.